codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Exception.java

```diff
@@ -47,6 +63,13 @@ public Http2Error error() {
     }
 
     /**
+     * If a connection error should attempt a graceful shutdown.
+     * @return {@code true} if a connection error should attempt a graceful shutdown. {@code false} otherwise.
+     */
+    public boolean gracefulShutdown() {
```

Comment:
@nmittler - Please note this change (and how it is used). A unit test was failing when the stream ids were exhausted...which lead me to look at this code path...and I realized in this case we should be attempting a graceful shutdown.  I could have done an `instanceof` check but figured it was more general to put support for this in `Http2Exception` (not this is not supported and always false for `StreamError`s.